server/hotrod/src/main/java/org/infinispan/server/hotrod/KeyValueRetrieverConverter.java

```diff
@@ -0,0 +1,86 @@
+package org.infinispan.server.hotrod;
+
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Set;
+
+import org.infinispan.commons.io.UnsignedNumeric;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.container.versioning.NumericVersion;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.notifications.cachelistener.filter.CacheEventConverter;
+import org.infinispan.notifications.cachelistener.filter.EventType;
+
+class KeyValueRetrieverConverter implements CacheEventConverter<byte[], byte[], byte[]> {
```

Comment:
I think we could reuse the existing converter rather than creating a new one that is very similar. 
The idea is to add a converter parameter to indicate if previous values should be returned, the default is ```false``` (the current behaviour). The Spring session would set it to ```true```.

Here's an example of a parameterised converter: https://github.com/infinispan/infinispan/blob/313b19301055c6267c6f2ea9065a7ab1b68099fe/server/integration/testsuite/src/test/java/org/infinispan/server/test/client/hotrod/CustomPojoEventConverterFactory.java

In the client side, you could set the parameter with:
```java
remoteCache.addClientListener(eventListener, null, new Object[]{Boolean.TRUE});
```

Reply:
👍

Reply:
nice, doing this

Reply:
as we discussed, this parameter stuff is a bit broken, so I did something very simple to grab the old value as well